### PR TITLE
[dong] support show thumbnail/embeded link - campaign show QR, info

### PR DESCRIPTION
**Pull request by phongvu2812**
# Ticket
 #535

# General
- [ ] No unnecessary files, debug logs, or commented code included.
- [ ] PR has at least 1 peer approval before senior review.
- [ ] I fully understand the code I submitted.
- [ ] No hardcoded configuration or secrets.
- [ ] **Environment Variables** changes? (Leave unchecked if N/A)

# Design & Solution Clarity
- [ ] A design diagram (sequence / flow) is attached or linked.
- [ ] The implementation matches the design.
- [ ] All major flows and decision points are represented in the diagram.
- [ ] Error handling and edge cases are clearly defined in the design.

Design link (diagram / doc):
`<paste link here>`

# Testing
- [ ] Unit tests cover main logic and important edge cases.

> Note: Complex logic must not rely on manual testing only.

# UI evidence (For frontend only)
`<paste screenshots here>`

# Peer Review Checklist
- [ ] I understand the overall solution and flow.
- [ ] The code follows the provided design diagram.
- [ ] No obvious logic or security issues found.
- [ ] Tests are sufficient for the scope of this change.

### CODE REVIEW

File: frontend/app/layout.tsx [hunk 1]
```diff
@@ -14,7 +14,22 @@ const inter = Inter({
   display: 'swap',
 });
 
+function getMetadataBase(): URL | undefined {
+  let raw = process.env.NEXT_PUBLIC_CHAT_APP_MMN_API_URL;
+  if (!raw) return undefined;
+  if (!raw.startsWith('http://') && !raw.startsWith('https://')) {
```
Comment (vietdangquoc) [inline frontend/app/layout.tsx:20]: if (!/^[a-zA-Z][a-zA-Z\d+\-.]*:\/\//.test(raw)) {
    raw = `https://${raw}`;
  }

**Review by hoangdohuy — APPROVED** [review]
LGTM
